### PR TITLE
Change FTL database permission to 664

### PR DESCRIPTION
**Pull request by yubiuser**
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** _{replace this text with a number from 1 to 10, with 1 being not familiar, and 10 being very familiar}_

1

---
Sets the file permission of the `pihole-FTL.db` to `664` to allow write permission to the `pihole` group. This is necessary for https://github.com/pi-hole/AdminLTE/pull/1886 as it allows the web interface write permission to delete items from the `message` table

### CODE REVIEW

**Comment by yubiuser** [comment]
I've been thinking about how to change permissions on already existing databases. My idea would be to move the whole section 

https://github.com/pi-hole/FTL/blob/b90ab8b1e78d7aece11c8fee1c3c00817761de52/src/database/common.c#L172-L175

from `db_create` to `db_init` right before the database is going to be opened

https://github.com/pi-hole/FTL/blob/b90ab8b1e78d7aece11c8fee1c3c00817761de52/src/database/common.c#L211

This would set the permissions every time the database in initialized (same for newly created db or existing databases). What do you think?

**Comment by DL6ER** [comment]
Yes, this sounds like a workable solution. There should be no harm involved in (re)setting the permissions once on every FTL start. FTL may not be `root` at the point where you want to do the `chown`, however, this shouldn't be a concern here because FTL has `CAP_CHOWN` which should give it the power of changing ownership of files even when not `root`.

**edit 1**

Thinking again about this, we might as well want to do it here:

https://github.com/pi-hole/pi-hole/blob/2673c2c0720fb3871a4c3cae6a3f38f9ccede24e/advanced/Templates/pihole-FTL.service#L29

because we never ensure the permissions are right after the file has been created.

**Comment by yubiuser** [comment]
Like that https://github.com/pi-hole/pi-hole/pull/4328?

File: src/database/common.c [hunk 1]
```diff
@@ -261,6 +256,11 @@ void db_init(void)
 		}
 	}
 
+		// Explicitly set permissions to 0664
+		// 664 =            u+w       u+r       g+w       g+r       o+r
+		const mode_t mode = S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH;
+		chmod_file(FTLfiles.FTL_db, mode);
+
```
Comment (DL6ER) [inline src/database/common.c:263]: Sorry, one last but, then we are ready to merge: Indentation is broken here.
Screenshot from 2021-09-23 20-51-52
```suggestion
	// Explicitly set permissions to 0664
	// 664 =            u+w       u+r       g+w       g+r       o+r
	const mode_t mode = S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH;
	chmod_file(FTLfiles.FTL_db, mode);

```

**Review by DL6ER — DISMISSED** [review]

**Review by DL6ER — APPROVED** [review]

**Comment by DL6ER** [comment]
Signed commits locally and updated the fork branch as discussed with @yubiuser

**Pull request merged**